src/components/admin/services/timer/Timer.jsx

```diff
@@ -0,0 +1,177 @@
+import { useState, useEffect, useRef } from "react";
+import TimerDisplay from "./TimerDisplay";
+import TimerStatus from "./TimerStatus";
+import TimerControls from "./TimerControls";
+import TimerEditMode from "./TimerEditMode";
+import { Progress } from "@/components/ui/progress";
+import { ChevronDown, ChevronUp } from "lucide-react";
+
+export default function Timer({ name, onRemove }) {
+  const [paused, setPaused] = useState(true);
+  const [total, setTotal] = useState(0);
+  const [isComplete, setIsComplete] = useState(false);
+  const [isEditMode, setEditMode] = useState(false);
+  const [original, setOriginal] = useState(0);
+  const [editedValue, setEditedValue] = useState("");
+  const [invalid, setInvalid] = useState(true);
+  const [collapsed, setCollapsed] = useState(false);
+
+  const timerRef = useRef(null);
+
+  const [time, setTime] = useState({
+    hours: 0,
+    minutes: 0,
+    seconds: 0,
+  });
+
+  useEffect(() => {
+    resetTimer();
+    setEditedValue("");
+  }, [isEditMode]);
+
+  useEffect(() => {
+    const calculateTime = (seconds) => {
+      return {
+        hours: Math.floor((seconds % 86400) / 3600),
+        minutes: Math.floor((seconds % 3600) / 60),
+        seconds: seconds % 60,
+      };
+    };
+
+    setTime(calculateTime(total));
+
+    if (total === 0) {
+      setPaused(true);
+      setIsComplete(true);
+      return;
+    }
+    if (isComplete) {
+      return;
+    }
+    if (paused) {
+      return;
+    }
+
+    timerRef.current = setTimeout(() => {
+      setTotal((prevSeconds) => prevSeconds - 1);
+    }, 1000);
+
+    return () => clearTimeout(timerRef.current);
+  }, [paused, isComplete, total]);
+
+  const pauseTimer = () => {
+    setPaused(true);
+  };
+
+  const resumeTimer = () => {
+    if (isComplete) {
+      setPaused(true);
+    } else {
+      setPaused(false);
+    }
+  };
+
+  const resetTimer = () => {
+    setTotal(original);
+    setIsComplete(false);
+    setPaused(true);
+  };
+
+  const openEditMode = () => {
+    setEditMode(true);
+    setEditedValue("");
+    setInvalid(true);
+  };
+
+  const numberToDate = (num) => {
+    const dateString = num.toString().padEnd(6, "0");
+    return {
+      hours: parseInt(dateString.substr(0, 2)),
+      minutes: parseInt(dateString.substr(2, 2)),
+      seconds: parseInt(dateString.substr(4, 2)),
+    };
+  };
+
+  const saveChanges = () => {
+    const dateObject = numberToDate(editedValue);
+
+    if (dateObject.minutes > 59 || dateObject.seconds > 59) {
+      setInvalid(true);
```

Comment:
instead of invalid, use a toaster to indicate the user must put a valid number, and then return from the function. we can also allow the users to put 99 seconds since it shouldn't affect any code logic hopefully